### PR TITLE
Authorization

### LINKED ISSUE #11: Skip to homepage if user is logged in
If a user logs in, then closes the page and reopens it, it will send them to the index page to login instead of recognizing they are already logged in and taking them straight to the homepage.

### PR DESCRIPTION
**Pull request by SeanT18**
This allows the user to login when entering the website. The format of the buttons are slightly off but the functionality is still there.

### CODE REVIEW

File: src/main/webapp/index.html [hunk 1]
```diff
@@ -14,5 +14,9 @@
             <div class="title">houseHOLD</div>
             <h3 class="body-font">Please log in here:</h3>
         </div>
+       <form action="/login" method="GET">
+            <button onclick="userLogin()" name="loginBtn">login button</button>
+            <button onclick="userLogin()" name="homeBtn">go to homepage</button>
+        </form>
```
Comment (apoorvadhanala) [inline src/main/webapp/index.html]: nit: indent -1

File: src/main/java/com/google/sps/servlets/loginServlet.java [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+        if (userService.isUserLoggedIn()) {
+            String userEmail = userService.getCurrentUser().getEmail();
+            String urlToRedirectToAfterUserLogsOut = "/";
+            String logoutUrl = userService.createLogoutURL(urlToRedirectToAfterUserLogsOut);
+            response.sendRedirect("/home.html");      
+        } else {
+            String urlToRedirectToAfterUserLogsIn = "/";
+            String loginUrl = userService.createLoginURL(urlToRedirectToAfterUserLogsIn);
+            response.sendRedirect(loginUrl);
+    }
```
Comment (apoorvadhanala) [inline src/main/java/com/google/sps/servlets/loginServlet.java:42]: fix indentation of parantheses

**Review by anthonyqzhu — CHANGES_REQUESTED** [review]
Change line 46 in loginServlet.java to "/home.html"

File: src/main/java/com/google/sps/servlets/loginServlet.java [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+
+    //checks if login was pressed
+    } else if (request.getParameter("loginBtn") != null) {
+        if (userService.isUserLoggedIn()) {
+            String userEmail = userService.getCurrentUser().getEmail();
+            String urlToRedirectToAfterUserLogsOut = "/";
+            String logoutUrl = userService.createLogoutURL(urlToRedirectToAfterUserLogsOut);
+            response.sendRedirect("/home.html");      
+        } else {
+            String urlToRedirectToAfterUserLogsIn = "/";
```
Comment (anthonyqzhu) [inline src/main/java/com/google/sps/servlets/loginServlet.java]: Change to /home.html

**Review by anthonyqzhu — APPROVED** [review]

**Comment by SeanT18** [comment]
Currently has a bug where if user is already signed in, it will still direct them to this page and than the homepage once the login button is clicked. Also UI position needs changing

**Review by apoorvadhanala — APPROVED** [review]

File: src/main/webapp/style.css [hunk 4]
```diff
@@ -105,3 +104,10 @@ body {}
   height: 75px;
   border:  thick solid rgb(205, 236, 232);
 }
+
+.login-button {
+     position:absolute;
```
Comment (yjennywang) [inline src/main/webapp/style.css:109]: indentation?

File: src/main/java/com/google/sps/servlets/loginServlet.java [hunk 5]
```diff
@@ -0,0 +1,44 @@
...
+
+@WebServlet("/login")
+public class loginServlet extends HttpServlet {
+
+  @Override
+  public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    UserService userService = UserServiceFactory.getUserService();
+    //TODO: redirect user to homepage if already logged in
+    response.setContentType("text/html");
+        if (userService.isUserLoggedIn()) {
```
Comment (yjennywang) [inline src/main/java/com/google/sps/servlets/loginServlet.java:33]: indentation -4

File: src/main/java/com/google/sps/servlets/loginServlet.java [hunk 6]
```diff
@@ -0,0 +1,44 @@
...
+import javax.servlet.annotation.WebServlet;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+@WebServlet("/login")
+public class loginServlet extends HttpServlet {
+
+  @Override
+  public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
```
Comment (yjennywang) [inline src/main/java/com/google/sps/servlets/loginServlet.java:29]: this should be a doPost.

doGets should only be used for things that don't change state

File: src/main/webapp/index.html [hunk 7]
```diff
@@ -12,7 +12,13 @@
         <div id="content">
             <br></br><br></br><br></br>
             <div class="title">houseHOLD</div>
-            <h3 class="body-font">Please log in here:</h3>
+                <h3 class="body-font">Please log in here:</h3>
```
Comment (yjennywang) [inline src/main/webapp/index.html:15]: can you check the indentation here?

File: src/main/webapp/script.js [hunk 8]
```diff
@@ -14,3 +14,10 @@
 
 
 
+function userLogin() {
+  fetch('/login').then(response => response.text())
+  .then((message) => {
```
Comment (yjennywang) [inline src/main/webapp/script.js:19]: at some point, consider adding error handling.

what happens if the server times out?

File: src/main/webapp/index.html [hunk 9]
```diff
@@ -12,7 +12,13 @@
         <div id="content">
             <br></br><br></br><br></br>
             <div class="title">houseHOLD</div>
-            <h3 class="body-font">Please log in here:</h3>
+                <h3 class="body-font">Please log in here:</h3>    
+            </div>
+            <div class="login-button">
+                <form action="/login" method="GET">
```
Comment (yjennywang) [inline src/main/webapp/index.html:18]: do you actually need the form action?

I think that only applies if there is a submit, right?

File: src/main/java/com/google/sps/servlets/loginServlet.java [hunk 10]
```diff
@@ -0,0 +1,44 @@
...
+    UserService userService = UserServiceFactory.getUserService();
+    //TODO: redirect user to homepage if already logged in
+    response.setContentType("text/html");
+        if (userService.isUserLoggedIn()) {
+            String userEmail = userService.getCurrentUser().getEmail();
+            String urlToRedirectToAfterUserLogsOut = "/";
+            String logoutUrl = userService.createLogoutURL(urlToRedirectToAfterUserLogsOut);
+            response.sendRedirect("/home.html");      
+        } else {
+            String urlToRedirectToAfterUserLogsIn = "/home.html";
```
Comment (yjennywang) [inline src/main/java/com/google/sps/servlets/loginServlet.java:39]: make this and urlToRedirectToAfterUserLogsOut a constant, or else just inline

File: src/main/java/com/google/sps/servlets/loginServlet.java [hunk 11]
```diff
@@ -0,0 +1,44 @@
...
+
+  @Override
+  public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    UserService userService = UserServiceFactory.getUserService();
+    //TODO: redirect user to homepage if already logged in
+    response.setContentType("text/html");
+        if (userService.isUserLoggedIn()) {
+            String userEmail = userService.getCurrentUser().getEmail();
+            String urlToRedirectToAfterUserLogsOut = "/";
+            String logoutUrl = userService.createLogoutURL(urlToRedirectToAfterUserLogsOut);
```
Comment (yjennywang) [inline src/main/java/com/google/sps/servlets/loginServlet.java:36]: Is the logoutUrl actually used anywhere?

**Pull request merged**